### PR TITLE
Added Session Present(LSB 0 bit) in CONNACK for MQTT v3.1.1 support

### PR DESCRIPTION
**Pull request by imsbhatnagar**

### CODE REVIEW

File: mqtt_test.go [hunk 1]
```diff
@@ -82,13 +82,14 @@ func TestEncodeDecode(t *testing.T) {
...
 			Msg: &ConnAck{
+				SessionPresent:true,
 				ReturnCode: RetCodeBadUsernameOrPassword,
 			},
 			Expected: gbt.InOrder{
 				gbt.Named{"Header byte", gbt.Literal{0x20}},
 				gbt.Named{"Remaining length", gbt.Literal{2}},
 
-				gbt.Named{"Reserved byte", gbt.Literal{0}},
+				gbt.Named{"Reserved byte", gbt.Literal{1}},
```
Comment (huin) [inline mqtt_test.go:92]: Just a nitty detail: could "Reserved byte" be replaced by "Connect Acknowledge Flags"?

This seems to be the name given here: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349253

**Review by huin — CHANGES_REQUESTED** [review]
Sorry for the delay. Looks good, just one small nit in the test.

File: mqtt_test.go (hunk 1 above)
Comment (imsbhatnagar) [reply mqtt_test.go]: agree, done and PR is updated.

**Review by huin — APPROVED** [review]
Thanks - approved!

**Pull request merged**